tests/Functional/ContentPublicationRenderTest.php

```diff
@@ -364,6 +367,85 @@ public function testPublicationRendering(): void {
     $content = $this->assertSession()->elementExists('css', '.ecl-row.ecl-u-mt-l .ecl-col-l-9');
     $content_items = $content->findAll('xpath', '/div');
     $this->assertCount(3, $content_items);
+
+    // Create publication collections that will reference the original
+    // publication.
+    $collection = $this->getStorage('node')->create([
+      'type' => 'oe_publication',
+      'title' => 'Test Publication collection node 1',
+      'oe_publication_type' => 'http://publications.europa.eu/resource/authority/resource-type/ABSTRACT_JUR',
+      'oe_publication_collection' => 1,
+      'oe_publication_publications' => [
+        'target_id' => $node->id(),
+      ],
+      'oe_publication_date' => [
+        'value' => '2020-04-15',
+      ],
+      'oe_subject' => 'http://data.europa.eu/uxp/1000',
+      'oe_author' => 'http://publications.europa.eu/resource/authority/corporate-body/AASM',
+      'oe_content_content_owner' => 'http://publications.europa.eu/resource/authority/corporate-body/COMMU',
+      'uid' => 0,
+      'status' => 1,
+    ]);
+    $collection->save();
+    $this->drupalGet($collection->toUrl());
+
+    // Assert the referenced publication is rendered as teaser.
+    $inpage_nav_expected_values['list'] = [
+      ['label' => 'Details', 'href' => '#details'],
+      ['label' => 'Documents', 'href' => '#documents'],
+    ];
+    $inpage_nav_assert->assertPattern($inpage_nav_expected_values, $navigation->getOuterHtml());
+
+    $assert = new ListItemAssert();
+    $content = $this->assertSession()->elementExists('css', '.ecl-row.ecl-u-mt-l .ecl-col-l-9');
+    $expected_values = [
+      'title' => 'Test Publication node',
+      'meta' => "Abstract, Legislative acts | 15 April 2020\n | Associated African States and Madagascar",
+      'description' => 'Test teaser text.',
+    ];
+    $assert->assertPattern($expected_values, $content->getOuterHtml());
+
+    // Assert the referenced publication page 'Part of collection' extra field.
+    $this->drupalGet($node->toUrl());
+    $details_expected_values['items'][0] = [
+      'label' => 'Part of collection',
+      'body' => 'Test Publication collection node 1',
+    ];
+    $first_element = [
+      'label' => 'Identification',
+      'body' => 'ID 1, ID 2',
+    ];
+    array_unshift($details_expected_values['items'], $first_element);
```

Comment:
```suggestion
    $details_expected_values = [
```
let's define `$details_expected_values` again, currently it is very hard to read and understand why we need array_unshift for.